src/docs/compact.dox

```diff
@@ -2,11 +2,49 @@
 
 The WT_SESSION::compact method can be used to compact data sources.
 
-The data source does not have to be quiescent, compaction may be
-performed on an active data source.
+The data source does not have to be quiescent, compaction may be performed on an active data source.
 
-Because checkpoints named by the application are not discarded until
-explicitly removed or replaced, they may prevent WT_SESSION::compact
-from accomplishing anything.
+<b>Note:</b> Because checkpoints named by the application are not discarded until explicitly removed or replaced, they may prevent WT_SESSION::compact from accomplishing anything.
+
+@section foreground Foreground compaction
+
+Foreground compaction is executed when a specific table is given to the WT_SESSION::compact method.
+The call returns once compaction has been executed on the table.
+
+Example:
+@code
+# Execute compaction on "mytable" with the default parameters.
+int ret = session->compact(session, "table:mytable", NULL);
+
+# Execute compaction on "mytable" if at least 20MB of free space can be theoretically reclaimed by compaction.
+int ret = session->compact(session, "table:mytable", "free_space_target=20MB");
+@endcode
+
+@section background Background compaction
+
+It is possible to interact with background compaction through the configuration item "background".
```

Comment:
This sentence feels like it is not completed. How about adding "background is provided to the WT_SESSION::compact method"?

Reply:
Thank you, I reworded in 2d3836de9cf37abd9ab8e6f9cc092568b0d67ca1, let me know what you think.